icu4j/main/tests/translit/src/com/ibm/icu/dev/test/translit/TransliteratorTest.java

```diff
@@ -2585,6 +2585,16 @@ public void TestScriptAllCodepoints(){
             }
         }
     }
+    
+    @Test
+    public void TestResourceBundleCrash() {
+        ULocale oldDefaultLocale = ULocale.getDefault();
+		ULocale.setDefault(ULocale.forLanguageTag("nb"));
```

Comment:
Suggestion 1: also tests for "nb", "ht" (maps to "fr-HT") and "hi-Latn" (maps to "en-IN").
Maybe "mo", in case it ever comes back :-)
Might be an array and iterate, I don't think it matters which one fails, as long as we detect a regression.

Suggestion 2: also test `Transliterator.getDisplayName("Any-Latin", ULocale.GERMAN);`
The second locale can be anything that does not have a .res file, so no ENGLISH or US.
Can be the same as default, but I used GERMAN to show that it is not in the same class with the failing locales
(where the parent.language <> getDefault.language)

Reply:
I decided to move the test into ICUResourceBundleTest, since the thing that's failing is the resource bundle-- it doesn't have anything to do with the transliterator, and it fails before we even try to get a specific resource out of the resource bundle.

I tried the locale IDs you suggested above: `nb` and `nn` fail, as we know.  `ht` did too.  For some reason, `hi-Latn` doesn't-- I'm not sure why, but it might be because the script code is changing the lookup pattern.  `mo` also didn't cause any problems.  But I changed the unit test to test all of these.

Reply:
> For some reason, hi-Latn doesn't-- I'm not sure why

I think it is because it maps to en-IN, and there is an `en.res` file.
This is one of the experiments I did. When I deleted that res file then `hi-Latn` also failed.

Reply:
> `mo` also didn't cause any problems.

Yes, it does not cause any problems today.

I know that the treatment of `mo` changed several times in the last 10 years.
As an experiment I've added a `ro` as a parent of `mo`, and then `mo` failed.

So the suggestion to add it to the test was as a abuse of caution :-), to detect a possible addition of `mo` => `ro`

Reply:
LGTM.

I will approve, and if you want to also add a a test for `Transliterator.getDisplayName("Any-Latin", ...);` fine.
If not, that's fine too, no need to explain why not :-)

Thanks,
Mihai